core/commonMain/src/kotlinx/serialization/encoding/Encoding.kt

```diff
@@ -378,12 +401,48 @@ public interface CompositeEncoder {
      * The element at the given [index] should have [PrimitiveKind.DOUBLE] kind.
      */
     public fun encodeDoubleElement(descriptor: SerialDescriptor, index: Int, value: Double)
+
     /**
      * Encodes a string [value] associated with an element at the given [index] in [serial descriptor][descriptor].
      * The element at the given [index] should have [PrimitiveKind.STRING] kind.
      */
     public fun encodeStringElement(descriptor: SerialDescriptor, index: Int, value: String)
 
+    /**
+     * Returns [Encoder] for decoding an underlying type of an inline class.
+     * [inlineDescriptor] describes a serializable inline class.
+     *
+     * Namely, for the `@Serializable inline class MyInt(val my: Int)`,
+     * and `@Serializable class MyData(val myInt: MyInt)`
+     * the following sequence is used:
+     * ```
+     * thisEncoder.encodeInlineElement(MyData.serializer.descriptor, 0, MyInt.serializer().descriptor).encodeInt(my)
+     * ```
+     *
+     * This method is an optimization and its invocation should have the exact same result as
+     * ```
+     * thisEncoder.encodeSerializableElement(MyData.serializer.descriptor, 0, MyInt.serializer(), myInt)
+     * ```
+     *
+     * Current decoder may return any other instance of [Encoder] class,
+     * depending on provided [inlineDescriptor].
+     * For example, when this function is called on Json encoder with
+     * `UInt.serializer().descriptor`, the returned encoder is able
+     * to encode unsigned integers.
+     *
+     * Note that this function returns [Encoder] instead of [CompositeEncoder]
+     * because inline classes always have one property.
+     * Calling [Encoder.beginStructure] on returned instance leads to an undefined behavior.
+     *
+     * @see Encoder.encodeInline
+     */
+    @ExperimentalSerializationApi
+    public fun encodeInlineElement(
+        descriptor: SerialDescriptor,
+        index: Int,
+        inlineDescriptor: SerialDescriptor
+    ): Encoder
```

Comment:
I've studied all our implementations of this method (end `decodeInlineElement` as well) and have the following observation:

It seems the third parameter is redundant and is always equal to `descriptor.getElementDescriptor(index)`.
Also, it's pretty error-prone for external serializers, because e.g. for `class CustomlySerializedClass(val color: Color)` `descriptor.getElementDescriptor(index)` and `Color.serializer().descriptor` may point to **different** descriptors (e.g. "hex string" and "single integer").

Can we get rid of the third parameter?